src/onegov/search/mixins.py

```diff
@@ -134,6 +134,122 @@ def es_tags(self):
         """ Returns a list of tags associated with this content. """
         return None
 
+    @property
+    def search_score(self):
+        """
+        the lower the score they higher the class type will be shown in search
+        results. Default is 10 (lowest)
+        """
+        return 10
+
+    def __lt__(self, other):
+        return self.search_score < other.search_score
+
+    @staticmethod
+    def psql_tsvector_string():
+        """
+        Provides the tsvector string for postgres defining which columns and
+        json fields to be used for full text search index.
+
+        Example:
+        s = create_tsvector_string('title', 'location')
+        s += " || ' ' || coalesce(((content ->> 'description')), '')"
+        s += " || ' ' || coalesce(((content ->> 'organizer')), '')"
+
+        :return: tsvector string
+        """
+        raise NotImplementedError
```

Comment:
See my comment on `Activity`. It seems to me like we should be able to generate the `tsvector_string` entirely based on what was previously available in terms of `es_` properties.

I technically don't mind if we move this complexity to the models, but then we need to be able to merge conflicting tsvectors into one that adds cases based on the polymorphic identity.

Reply:
I tried using `hasattr` and `getattr` and `isinstance` but the `content_property()`s are special not real attributes. Also the type information cannot be used to distinguish beetween meta and content. Any idea how to identify if the property is in meta or content column?
```
(Pdb) self
<class 'onegov.org.models.page.Topic'>
(Pdb) getattr(self, 'title')
<sqlalchemy.orm.attributes.InstrumentedAttribute object at 0x7f485f7e58f0>
(Pdb) getattr(self, 'lead', None) == None
True
``

Below the snipped of the content_property lead
```class Topic(Page, TraitInfo, SearchableContent, AccessExtension,
            PublicationExtension, VisibleOnHomepageExtension,
            ContactExtension, ContactHiddenOnPageExtension,
            PersonLinkExtension, CoordinatesExtension, ImageExtension):
    __mapper_args__ = {'polymorphic_identity': 'topic'}

    es_type_name = 'topics'

    lead = content_property()
``

Reply:
`content_property` and `meta_property` are just factory functions which create a `onegov.core.orm.mixins.content.dict_property` so in this example you would be able to look at `cls.lead.attribute` and see that it is set to `content`, which would tell you that the value is stored inside `content`.

Reply:
The other complication is that there's some magic in the Declarative ORM metaclass, you probably have to look at the mapper to get the real property, i.e. 
```python
from sqlalchemy import inspect
getattr(inspect(cls), 'lead', None)
```

Reply:
I managed to bring all tsvector code to mixins except one